compiler-core/src/type_/environment.rs

```diff
@@ -241,6 +245,45 @@ impl<'a, 'b> Environment<'a, 'b> {
         }
     }
 
+    /// Lookup constructors for type in the current scope.
+    ///
+    pub fn get_constructors_for_type(
+        &self,
+        module_alias: &Option<String>,
+        name: &str,
+    ) -> Result<&Vec<String>, UnknownTypeConstructorError> {
```

Comment:
@lpil Can I make this assume success? (I think this will be used always after unifying types and confirming the type is there)

Reply:
I think this function should return a result because the Environment doesn't know anything about when it is being used. We would assert value at present at a higher level of abstraction in the typer.